### DISCUSSION TITLE
Filter documents which *not* belong to a set of tags.

### DISCUSSION THREAD

**Discussion by tobi-bo** [jonaswinkler/paperless-ng#401] [Feature Requests]
First, this is a great project, I started to use it some days ago and imported lots of documents so far. It works great and reliable.

One thing, that I would like to do, is filter documents for tags, which they do **not** have. 
To clarify, the use case is as follows: I have certain tags in different categories (identified by colors, which is a nice feature by the way). For example I have green tags, to identify the person a document belongs to, for example "me", "my wife", etc. Another category of tags (gray) is the source of the document, for example "email", "by post" or something like this and there are more "categories". Now I would like to show all documents which are not tagged with a person yet (green category) to probably add one. For that I would like to list all documents which are not tagged with "me" or "my wife". I cannot use the "none" filter, because almost all of the documents are tagged with something from a different category like "email".

There are two derived workflows:

- find all documents which need a tag from a specific category.
- find new tags which fill gaps in a specific category.

I hope it is clear what I mean, please let me know otherwise ;-)

I figured out that I can use a search term like `NOT tag:"me" AND NOT tag:"my wife"`, but as far as I can see, I cant add search queries as a saved view.

Is it possible to filter documents like this or to add any (complex) search query to the saved views?

Any hint is highly appreciated.

**Comment by jonaswinkler** [comment]
Hello, and thank you for your feedback!

Please have a quick look at how label filtering works on issue lists on GitHub: It's possible to both filter for issues that have a certain label and for issues that do not have a certain label by alt-clicking labels. Would something like this fit your needs? (Actually, the fitlering interface in paperless is heavily inspired by how it works on the issue list).

Internally, search results work *very* different from how filtering works on the document list, and I cannot provide saved views for that.

**Reply by tobi-bo** [reply]
Hi,

> It's possible to both filter for issues that have a certain label and for issues that do not have a certain label by alt-clicking labels. Would something like this fit your needs?

Yes, this is exactly what I mean.

**Reply by sbrunner** [reply]
We can have a workaround by using a tag:
e.-g.: for document that match A but not B
- Create a custom tag names advance-search
- Configure the filter A
- Set to all documents the tag advance-search
- Search all document that match the tag advance-search and B
- Unset to all documents the tag advance-search
- Search all document that match the tag advance-search
A bit long but you get what you want :-)

**Comment by shamoon** [comment]
#399 has this (if it gets merged TBD)

**Reply by amo13** [reply]
Awesome, I was just looking for the exact features you have in the prepared pull request! Hopefully it will be merged at some time :)
In the meantime, I'm glad this is already there hidden by typing `NOT tag:"me"` as mentioned above. That's making my filtering needs a whole lot easier.
